lib/core/CJsonStateRestoreTraverser.cc

```diff
@@ -349,96 +395,133 @@ bool CJsonStateRestoreTraverser::SRapidJsonHandler::Bool(bool b) {
+    LOG_TRACE(<< "on_string_part: '" << s << "', n: " << n);
+
+    s_Type = E_TokenStringPart;
     if (s_RememberValue) {
-        s_Value[s_NextIndex].assign(CStringUtils::typeToString(u));
+        if (m_NewToken) {
+            s_Value[s_NextIndex].clear();
+            m_NewToken = false;
+        }
+        s_Value[s_NextIndex].push_back(s.front());
     }
 
     return true;
 }
 
-bool CJsonStateRestoreTraverser::SRapidJsonHandler::Double(double d) {
-    s_Type = E_TokenDouble;
-    if (s_RememberValue) {
-        s_Value[s_NextIndex].assign(CStringUtils::typeToString(d));
+bool CJsonStateRestoreTraverser::SBoostJsonHandler::on_string( std::string_view s, std::size_t n, json::error_code&/* ec*/ ) {
+    if (s_Type != E_TokenStringPart && s.front() == '"') { // Empty string
+        s_Value[s_NextIndex].clear();
     }
+    LOG_DEBUG(<< "on_string: '" << s_Value[s_NextIndex] << "', n: " << n << ", s_NextIndex: " << s_NextIndex);
 
+    s_Type = E_TokenString;
+    if (s_RememberValue) {
+        if (s.front() != '"') {
+            s_Value[s_NextIndex].push_back(s.front());
+        }
+        m_NewToken = true;
+    }
     return true;
 }
 
-bool CJsonStateRestoreTraverser::SRapidJsonHandler::RawNumber(const char*,
-                                                              rapidjson::SizeType,
-                                                              bool) {
-    return false;
+bool CJsonStateRestoreTraverser::SBoostJsonHandler::on_document_begin( json::error_code& ec ) {
+    LOG_DEBUG(<< "on_document_begin");
+
+    return (ec) ? false : true;
 }
 
-bool CJsonStateRestoreTraverser::SRapidJsonHandler::String(const char* str,
-                                                           rapidjson::SizeType length,
-                                                           bool) {
-    s_Type = E_TokenString;
-    if (s_RememberValue) {
-        s_Value[s_NextIndex].assign(str, length);
+bool CJsonStateRestoreTraverser::SBoostJsonHandler::on_object_begin(json::error_code& ec) {
+    LOG_DEBUG(<< "on_object_begin");
+    s_Type = E_TokenObjectStart;
+
+    if (ec) {
+        return false;
     }
 
+    if (s_RememberValue) {
+        ++s_Level[s_NextIndex];
+        s_Value[s_NextIndex].clear();
+    }
     return true;
 }
 
-bool CJsonStateRestoreTraverser::SRapidJsonHandler::StartObject() {
-    s_Type = E_TokenObjectStart;
+bool CJsonStateRestoreTraverser::SBoostJsonHandler::on_key_part( std::string_view s, std::size_t n, json::error_code& ec ) {
+    s_Type = E_TokenKeyPart;
+    LOG_TRACE(<< "on_key_part: '" << s << "', n: " << n);
+    if (ec) {
+        return false;
+    }
+
     if (s_RememberValue) {
-        ++s_Level[s_NextIndex];
-        s_Value[s_NextIndex].clear();
+        if (m_NewToken) {
+            m_NewToken = false;
+            s_NextIndex = 1 - s_NextIndex;
+            s_Name[s_NextIndex].clear();
+        }
+        s_Name[s_NextIndex].push_back(s.front());
```

Comment:
This would avoid propagating knowledge of how many characters were parsed in one go:

```suggestion
        s_Name[s_NextIndex].append(s);
```

Same in the other methods that are currently pushing `s.front()`.